### ISSUE TITLE
Controlled fields unsync after a validation

### ISSUE THREAD

**Issue by ivan-kleshnin** [react-hook-form/react-hook-form#1197]
**Describe the bug**

It's hard to describe in two sentences so I recorded a (short) video:
https://youtu.be/pGlLmBDLfNA

The gist of the bug (if that is a bug) is that two controlled fields

```js
let [state, setState] = useState({
    sync: true,
    title: "",
    seoTitle: "",
  })

  let {errors, handleSubmit, register} = useForm({
    title: "",
    seoTitle: "",
    // the rest of the form is UNCONTROLLED (generic)
  })

  ...

Sync: <input type="checkbox" checked={state.sync} onChange={e => setSync(e.target.checked)}/>

<TextInput name="title"
               label="Title"
               value={state.title}
               onChange={e => setTitle(e.target.value)}
               register={register}
               options={{required: true, minLength: 5, maxLength: 60}}
               error={errors.title}/>

  <TextInput name="seoTitle"
             label="SEO Title"
             value={state.seoTitle}
             onChange={e => setSeoTitle(e.target.value)}
             register={register}
             options={{required: true, minLength: 5, maxLength: 80}}
             error={errors.seoTitle}/>

// The rest of the form is UNCONTROLLED (generic)
```

become unsynchronized after a validation.

An example, mentioned in video: https://github.com/ivan-kleshnin/react-hook-form-bug/tree/master/src

**To Reproduce**

1. Clone the repo: https://github.com/ivan-kleshnin/react-hook-form-bug/tree/master/src
2. Input some text up to 3 chars
3. Submit, get an error
4. Keep inputting. See how fields are unsynchronized at this stage.

**Software**
```
"react-hook-form": "^5.0.3",
```

**Comment by bluebill1049** [comment]
what's the usage like? you are mix uncontrolled with controlled.

**Comment by ivan-kleshnin** [comment]
I don't really know what to add more: there's a video, a sandbox, a description.

**Comment by bluebill1049** [comment]
1. you raised as a bug and i tagged as not valid, because it's not a bug. I didn't close the issue which means I am here to help. 

2. Your time is as valuable as mine, in fact i am probably reading/solving/working more issues than just yours. In the template we labeled codesandbox as **required**, because i can check it and reproduce issue much easier. Can you imagine i have check out each issue with a Github repo... It's not hard to follow the template right, at the same i appreciated your video and detailed report, otherwise i would close it right away. Sometimes we should learn to look things at different perspective. 

> P.S. It's strange that you immediately added not-valid tag. I'm ok about the idea that it's not a bug but a wrong usage, but, you know, the number of open issues has nothing to do with project quality.

Please... this is completely unnecessary comments. Doesn't adds any value to this issue. How i maintain a lib it's up to me, I don't tell you how to do your job. In fact, i am doing it all for free as open source. 

---------------------------------------------------------------

Ok, Let's look at your questions.

1. if you working with external controlled component, use Controller: https://react-hook-form.com/api#Controller
2. if you want to read value from other inputs, use watch api: https://react-hook-form.com/api#watch

**Comment by ivan-kleshnin** [comment]
You're taking it too serious IMO. But, as you said, it's up to you. I'll figure out the solution myself.
